build-tools/packages/build-tools/src/repoPolicyCheck/handlers/fluidBuildTasks.ts

```diff
@@ -79,16 +79,30 @@ function getFluidPackageMap(root: string) {
  *
  * @param json - the package.json content to search script in
  * @param command - the command to find the script name for
- * @returns  first script name found to match the command
+ * @returns best script name found to match the command
  */
 function findScript(json: PackageJson, command: string) {
+	// Multiple scripts can have the same command, we want to find the best one.
+	let bestScript: { rank: number; script: string | undefined } = { rank: 0, script: undefined };
 	for (const script in json.scripts) {
 		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-		if (command === json.scripts[script]!) {
-			return script;
+		const scriptCommands = json.scripts[script]!.split("&&");
+		scriptCommands.forEach((scriptCommand, index) => {
+			if (command === scriptCommand.trim()) {
+				// Rank better (lower) when there are fewer commands and the command is earlier
+				// in the list.
+				const rank = (index + 1) * scriptCommands.length;
+				if (bestScript.script === undefined || rank < bestScript.rank) {
+					bestScript = { rank, script };
+				}
+			}
+		});
+		// If we find an exact match, we can stop looking.
+		if (bestScript.rank === 1) {
+			return bestScript.script;
 		}
 	}
-	return undefined;
+	return bestScript.script;
```

Comment:
Seams reasonablish...  I vaguely understand that FluidBuild is looking for tasks containing 'tsc' to build/check dependencies and assume the complication is that we're now using 'tsc && copyfiles'.

From your comments on the overall PR, I'm guessing the alternative would be to break 'tsc && copyfiles' into separate scripts/tasks with more scripts/tasks to tie them together.

I'm happy to trust your judgement on the tradeoffs.

Reply:
There was already some support for && in the dependencies but it didn't extend to the tsc searches.
Once I added support for && to that, I found that there were other packages with tsc commands embeded. packages/dd/matrix bench:profile is the notable one. AFAIK it is actually outside the general build:fast support and doesn't actually build like the cases this is trying to find.
I had added this comment to the description but didn't find a good place to have it in code:
- Recognize more multi-command scripts in package.json
  Ideally, we would use broken down scripts, but
  those result in additional dependency complexities
  and don't offer simple atomicity.